features/frameworks/greentea-client/greentea-client/test_env.h

```diff
@@ -28,6 +28,9 @@
 #endif
 
 #include <stdio.h>
+extern "C" {
```

Comment:
This hasn't really come up before. There has always been clear separation between C and C++ code. Question really is why this isn't the case now for greentea? I'd guess this has to do with using RawSerial and Tickers in the test harness?? @mazimkhan are you planning to bind greentea to mbed OS drivers or a different runtime for put/get?